### DISCUSSION TITLE
Is there a way to replay from a Batch run a node, without repeating the whole process?

### DISCUSSION THREAD

**Discussion by hoomank** [microsoft/promptflow#2886] [General]
I am looking for an efficient process to get visibility of why some nodes failed, for instance when doing a batch run, and to then correct the code in the failed nodes (or the ones supplying invalid inputs) until they stop failing. So far, what I do is use the "visualize run" feature which lets me quickly see what runs failed in a batch run. But once I know what node failed, I want to debug it. 

Is there any way to re-run the failed node without re-running the whole flow? This would allow one to keep tweaking that node (such as a python module) until it stops failing.

**Comment by D-W-** [comment]
Hi @hoomank , in `flow test`, you can test a single node without running the whole flow. Here's related doc: https://microsoft.github.io/promptflow/how-to-guides/develop-a-flow/init-and-test-a-flow.html#test-a-single-node-in-the-flow
